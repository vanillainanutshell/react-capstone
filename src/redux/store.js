import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const reducer = combineReducers({

});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;
