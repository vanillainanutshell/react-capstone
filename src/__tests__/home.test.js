import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Home from '../components/home';

describe('Check if all components are rendered', () => {
  test('Check header', () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(screen.getByText('Top Rated Animes')).toBeInTheDocument();
  });
});

describe('Sanpshoot of the Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
