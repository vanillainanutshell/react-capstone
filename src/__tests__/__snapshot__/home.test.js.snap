// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanpshoot of the Home renders correctly 1`] = `
<div>
  <div
    className="headline"
  >
    <img
      alt="knight"
      className="knight"
      src="knight.gif"
    />
    <h1>
      Current Top 10 AOE2:DE Players
    </h1>
    <img
      alt="knight"
      className="knight"
      src="knight.gif"
    />
  </div>
  <div
    className="stats_title_cnt"
  >
    <input
      onChange={[Function]}
      placeholder="Player name"
    />
  </div>
  <div
    className="content_cnt"
  />
</div>
`;
